smack-extensions/src/main/java/org/jivesoftware/smackx/mood/MoodListener.java

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mood;
+
+import org.jivesoftware.smackx.mood.element.MoodElement;
+
+import org.jxmpp.jid.BareJid;
+
+public interface MoodListener {
+
+    void onMoodUpdated(BareJid jid, MoodElement moodElement);
```

Comment:
Such listeners should also provide a reference to the Stanza which caused their invocation. In this case, add a third `Message` argument.